### PR TITLE
[codex] harden keeper detail source contracts

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
- move keeper detail source resolution into a dedicated selector layer
- harden dashboard types/normalizers so shell and mission compatibility fields are not treated as authored policy
- add tests for config preload, observed audit precedence, and duplicate field dropping

## Why
keeper detail had the main tool-policy boundary fixed, but type surfaces still allowed shell and mission projections to look like authoritative authored config. this kept the source contract partially implicit and easy to regress.

## Impact
- keeper detail keeps using `/api/v1/keepers/:name/config` as SSOT for authored/effective tool policy
- mission and shell projections remain wire-compatible, but the dashboard no longer promotes their compatibility fields into keeper-detail SSOT
- regressions are covered with focused dashboard tests

## Validation
- `./node_modules/.bin/tsc --noEmit --pretty false`
- `./node_modules/.bin/vitest run src/components/keeper-detail-runtime.test.ts src/components/keeper-detail.test.ts src/keeper-store-normalize.test.ts src/mission-normalizers-entities.test.ts src/components/keeper-config-panel.test.ts`
- `dune build --root /Users/dancer/me/workspace/yousleepwhen/masc-mcp/.worktrees/fix/keeper-detail-ssot-contracts`

### CODE REVIEW

**Comment by jeong-sik** [comment]
Cross-model review evidence:

- Reviewer: direct `sb glm-text` (ZAI GLM path)
- Reviewed commit: `f1f36eb6d`
- Reviewed at: 2026-04-02 Asia/Seoul
- Scope: keeper detail source-contract hardening diff only

Summary:
- GLM surfaced low-risk notes around removed compatibility fields and import wiring.
- I verified with repo grep that there are no remaining dashboard consumers of removed `DashboardMissionKeeperBrief.allowed_tool_names` or removed `Keeper` tool-policy duplicate fields.
- The new `keeper-detail.test.ts` mock path matches the real import path in `keeper-detail.ts`.
- Result: no actionable correctness findings for this diff.

File: dashboard/src/components/keeper-detail-source.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import { missionKeeperBriefs } from '../mission-store'
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply dashboard/src/components/keeper-detail-source.ts]: 동의합니다. 이미 커밋 ebe6493dd ("refactor(review): import keeper briefs from mission signals")에서 `mission-store` barrel 대신 `../mission-signals`에서 직접 import하도록 수정 완료했습니다.
